pipelines/controller/src/main/java/com/google/fhir/analytics/PipelineManager.java

```diff
@@ -104,16 +105,23 @@ public class PipelineManager implements ApplicationListener<ApplicationReadyEven
   void publishPipelineMetrics(MetricResults metricResults) {
     MetricQueryResults metricQueryResults = EtlUtils.getMetrics(metricResults);
     for (MetricResult<Long> metricCounterResult : metricQueryResults.getCounters()) {
-      meterRegistry.gauge(
-          metricCounterResult.getName().getNamespace()
-              + "_"
-              + metricCounterResult.getName().getName(),
-          metricCounterResult.getAttempted());
+      Gauge.builder(
+              metricCounterResult.getName().getNamespace()
+                  + "_"
+                  + metricCounterResult.getName().getName(),
+              () -> metricCounterResult.getAttempted())
+          // Make a strong reference so that the value is not immediately cleared after GC
+          .strongReference(true)
+          .register(meterRegistry);
```

Comment:
Without knowing much about the internals of micrometer, I suppose we won't have a memory leak issue by registering these metrics after each pipeline run; can you please confirm?